### PR TITLE
Add CSE 4.2.2 and 4.2.3 support

### LINKED ISSUE #1319: CSE 4.2.2/4.2.3 Compatiblity
Hi Team, are we safe to assume TFV 3.13.0 is compatible with CSE 4.2.2 as, basically, there are no funcionality changes between 4.2.1 and 4.2.2?

### PR DESCRIPTION
**Pull request by adambarreiro**
⚠️  **PR destination: [release/v3.x](https://github.com/vmware/terraform-provider-vcd/tree/release/v3.x) branch**

This PR adds support for CSE 4.2.2 and 4.2.3 in `vcd_cse_kubernetes_cluster` and updates installation guide and examples.

Tests:

- [x] Installed a fresh 4.2.2 and created a cluster
- [x] Installed a fresh 4.2.3 and created a cluster
- [x] Installed a fresh 4.2.1, upgrade to 4.2.2 and create a cluster
- [x] Installed a fresh 4.2.2, upgrade to 4.2.3 and create a cluster
- [x] Installed a fresh 4.2.1 and created a cluster

Closes #1319

### CODE REVIEW

File: vcd/resource_vcd_cse_kubernetes_cluster.go [hunk 1]
```diff
@@ -33,8 +33,7 @@ func resourceVcdCseKubernetesCluster() *schema.Resource {
 			"cse_version": {
 				Type:         schema.TypeString,
 				Required:     true,
-				ForceNew:     true,
```
Comment (adambarreiro) [inline vcd/resource_vcd_cse_kubernetes_cluster.go:36]: Removed as it is not needed

**Comment by adambarreiro** [comment]
Putting back to draft as CSE tests need an update

**Review by Didainius — APPROVED** [review]
I have tried out the build process and validate it resulted a working cluster on VCD 10.6.0.1 with 4.2.2 and 4.2.3

File: vcd/resource_vcd_cse_kubernetes_cluster.go (hunk 1 above)
Comment (lvirbalas) [reply vcd/resource_vcd_cse_kubernetes_cluster.go:36]: Interesting.

File: vcd/resource_vcd_cse_kubernetes_cluster_test.go [hunk 2]
```diff
@@ -746,6 +754,10 @@ func TestAccVcdCseKubernetesClusterFailure(t *testing.T) {
 	preTestChecks(t)
 	requireCseConfig(t, testConfig)
 
+	if testConfig.Cse.Version == "4.2.2" || testConfig.Cse.Version == "4.2.3" {
+		t.Skip("test does not work with System Administrator")
```
Comment (lvirbalas) [inline vcd/resource_vcd_cse_kubernetes_cluster_test.go:758]: These two error messages do not seem to match the if statement.

Comment (adambarreiro) [reply vcd/resource_vcd_cse_kubernetes_cluster_test.go:758]: Improved message

**Pull request merged**